### ISSUE TITLE
Problems with redundantObjc rule

### ISSUE THREAD

**Issue by davidelorenzitw** [nicklockwood/SwiftFormat#438]
We have issues with the `redundantObjc` rule, which is removing some @objc modifier to functions that need it to be able to be triggered as targets for NSNotifications or UIButton touches, for example.
This makes the code not compile anymore.

We are running SwiftFormat with a configuration file, where all the rules are enabled: only the aforementioned rule has been then disabled.

Simple snippet of original code - before running SwiftFormat:

```
@objcMembers
public final class SomeClass: NSObject {

private func someFunction() {
        NotificationCenter.default.addObserver(self,
                                               selector: #selector(handleNotification(_:)),
                                               name: UIResponder.keyboardWillShowNotification,
                                               object: nil)
}

@objc private func handleNotification(_ notification: NSNotification) { ... }
```

After running SwiftFormat, the `@objc` modifier of the `handleNotification` function is removed, leaving this error:

![Screenshot 2019-05-14 at 09 17 53](https://user-images.githubusercontent.com/41289135/57682313-73cc0d00-7629-11e9-8796-9dca9449f201.png)

I don't know if there is a way to be able to detect an @objc function that is used as a selector destination while parsing the file (at least within the file scope).
Maybe you could add an extra requirement that the rule, to be effective, only perform this check if the function is declared as private, thus ensuring that the scope where it can be accessed is limited.
IMO, it will also enforce better coding practices, not leaving such a function exposed/callable from other components.

**Comment by nicklockwood** [comment]
Thanks for reporting. I think the `private` requirement is a little too obscure since many developers wouldn't do that and would then still encounter the bug.

I'll think about how to solve this for the general case. In the meantime though, note that you can disable the rule for individual methods by putting

```swift
// swiftformat:disable:next redundantObjc
```

On the preceding line.

**Comment by nicklockwood** [comment]
Fixed in version 0.40.9.

For now I just disabled stripping `@objc` from private members, since that seems to be the only case which causes the compilation error. The idea about checking for the existence of `#selector`s for that specific function is a good one though, and I may revisit it.
